handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -564,7 +564,10 @@ public final void setCloseNotifyReadTimeoutMillis(long closeNotifyReadTimeoutMil
      * Returns the {@link SSLEngine} which is used by this handler.
      */
     public SSLEngine engine() {
-        return engine;
+        // If its an instanceof Java9SslEngineWrapper we need to unwrap it before returning it to the user
+        // as Java9SslEngineWrapper does not delegate all the new methods that are added to SSLEngine in Java9.
+        return engine instanceof Java9SslEngineWrapper ?
```

Comment:
discussed with @normanmaurer offline. This may lead to treating the SslSession object differently in JDK9 and lower ... can we avoid this?